docs/components/top_app_bars.md

```diff
@@ -0,0 +1,351 @@
+
+The following example shows a contextual action bar with a contextual title, a close icon, two contextual action icons, and an overflow menu:
+
+![Contextual top app bar with close button, download, garbage, and overflow menu icons](assets/top_app_bars/contextual_screenshot.png)
+
+```dart
+AppBar(
+    leading: Icon(Icons.close),
+    title: Text('1 selected'),
+    actions: <Widget>[
+        Icon(Icons.file_upload),
+        Padding(
+            padding: EdgeInsets.symmetric(horizontal: 16),
+            child: Icon(Icons.delete),
+        ),
+        Icon(Icons.more_vert),
+    ],
+    backgroundColor: Colors.black87,
+),
+
+```
+
+### Anatomy and Key properties
+
+![Contextual action bar anatomy diagram](assets/top_app_bars/contextual_action_bar_anatomy.png)
+
+1. Close Button (optional)
+1. Contextual title (optional)
+1. Contextual action items (optional)
+1. Overflow menu (optional)
+
+<b>Close Button (optional)</b>
+
+| &nbsp; | Property |
+| --- | --- |
+| **Icon** |  In `leading` parameter, use `IconButton` Widget |
+| **Color** | In `leading` parameter, use `IconButton` Widget and set parameter `color` |
+| **Size** | In `leading` parameter, use `IconButton` Widget and set parameter `iconSize` |
+| **Gravity** | `titleSpacing` parameter |
+| **Padding** | Wrap `Icon` wiget with `Padding` widget |
+
+<b>Contextual title (optional)</b>
+
+| &nbsp; | Property |
+| --- | --- |
+| **Text label** | In `title` parameter, use `Text` widget |
+| **Color** | In `title` parameter, use `Text` Widget and and set parameter `style` |
+| **Typography** | In `title` parameter, use `Text` Widget and and set parameter `style` |
+
+<b>Contextual action items (optional)</b>
+
+| &nbsp; | Property |
+| --- | --- |
+| **Icon** | In `action` parameter, use `IconButton` widget and set parameter `icon` |
+| **Color** | In `action` parameter, use `IconButton` widget and set parameter `color` OR set `actionIconsTheme` |
+| **Size** | In `action` parameter, use `IconButton` widget and set parameter `iconSize` OR set `actionIconsTheme`|
+| **Padding** | Wrap `IconButton` widget with `Padding` widget |
+
+<b>Overflow menu (optional)</b>
+
+| &nbsp; | Property |
+| --- | --- |
+| **Icon** | In `action` parameter, add `IconButton` widget to the end of Widget list and set parameter `icon` within `IconButton` |
+| **Color** | In `action` parameter, add `IconButton` widget to the end of Widget list and set parameter `color` within `IconButton` |
+| **Size** | In `action` parameter, add `IconButton` widget to the end of Widget list and set parameter `iconSize` within `IconButton` |
+| **Padding** | In `action` parameter, add `IconButton` widget to the end of Widget list and wrap that Widget with a `Padding` widget  |
+
+#### Styles
+
+_If the component API implements multiple types, include any style information that differentiates the types in a table_
+
```

Comment:
Styles doesn't seem to apply to flutter -- probably best delete this.

Reply:
Done